tests/cli/test_start_server.py

```diff
@@ -413,3 +414,16 @@ def test_choose_when_multiple_profiles_have_same_api_url(self):
 
         profiles = load_profiles()
         assert profiles.active_name == "local"
+
+
+class TestServerExitCodes:
+    def test_misconfigured_server_exits_nonzero(self) -> None:
+        invalid_connection_url = "FOOBAR!"
+        with temporary_settings(
+            {PREFECT_API_DATABASE_CONNECTION_URL: invalid_connection_url}
+        ):
+            invoke_and_assert(
+                command=["server", "start"],
+                expected_code=1,
+                expected_output_contains=invalid_connection_url,
+            )
```

Comment:
When I run this test locally, it hangs. It think the `temporary_settings` isn't propagating to the process running the server in this test. It works with this update:
```suggestion
    def test_misconfigured_server_exits_nonzero(self, monkeypatch) -> None:
        monkeypatch.setenv("PREFECT_API_DATABASE_CONNECTION_URL", "FOOBAR!")

        invoke_and_assert(
            command=["server", "start"],
            expected_code=1,
            expected_output_contains="FOOBAR!",
        )
```

Reply:
Implemented here [7756c13](https://github.com/PrefectHQ/prefect/pull/16476/commits/7756c1379224fe64926416d970dcc590d546003a)

All the tests pass for me locally (ubuntu 24.04). However, it seems like some test fails for whatever reason in the pipeline - I'm not sure what's going on there, but maybe you can advice?

Reply:
It might be better to update some existing tests in that case. If you update the tests in `TestUvicornSignalForwarding` in the same file to check the return code, then I think that covers the change well. You can do that by updating each place we wait for a server process to exit:
```python
await server_process.wait()
```
to verify the return code is non-zero:
```python
assert await server_process.wait() != 0, "Server process exited with zero code"
```
Assertions like that fail on `main` but pass with your PR.